syntax = "proto3";

package envoy.extensions.network.dns_resolver.apple.v3;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.network.dns_resolver.apple.v3";
option java_outer_classname = "AppleDnsResolverProto";
option java_multiple_files = true;
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: apple DNS resolver configuration]

// Configuration for apple DNS resolver
// For Envoy built in MacOS to use apple DNS library as DNS resolver, the following need to be done:
// 1) In file: :repo:`extensions_build_config.bzl <source/extensions/extensions_build_config.bzl>`,
// enable apple DNS resolver extension by adding below line
//      "envoy.dns_resolver.apple":                "//source/extensions/network/dns_resolver/apple:config",
// 2) In corresponding Envoy configuration, for example:
//    :repo:'bootstrap.proto <api/envoy/config/bootstrap/v3/bootstrap.proto>`,
// add apple DNS resolver as a typed extension in field:
//    :ref:'typed_dns_resolver_config <envoy_v3_api_field_config.bootstrap.v3.Bootstrap.typed_dns_resolver_config>'
//
// typed_dns_resolver_config:
//   name: envoy.dns_resolver.apple
//   typed_config:
//     "@type": type.googleapis.com/envoy.extensions.network.dns_resolver.apple.v3.AppleDnsResolverConfig
message AppleDnsResolverConfig {
}
